internal/encoder/compiler.go

```diff
@@ -506,8 +506,6 @@ func (c *Compiler) listElemCode(typ *runtime.Type) (Code, error) {
 
 func (c *Compiler) mapKeyCode(typ *runtime.Type) (Code, error) {
 	switch {
-	case c.implementsMarshalJSON(typ):
-		return c.marshalJSONCode(typ)
 	case c.implementsMarshalText(typ):
```

Comment:
Should we care about `MarshalText` ?

Reply:
I guess it's ok since MarshalText can't return anything but the string